cli/cmd/cluster.go

```diff
@@ -995,29 +993,51 @@ func updateNodeGroupScale(clusterConfig clusterconfig.Config, targetNg string, d
 		}
 		availableNodeGroups = append(availableNodeGroups, ng.Name)
 		if ng.Name == targetNg {
-			if ng.MinInstances == desiredMinReplicas && ng.MaxInstances == desiredMaxReplicas {
-				fmt.Printf("no changes to the %s nodegroup required in cluster %s from region %s\n", ng.Name, clusterName, region)
+			var minReplicas, maxReplicas int64
+			if desiredMinReplicas == nil {
+				minReplicas = ng.MinInstances
+			} else {
+				minReplicas = *desiredMinReplicas
+			}
+			if desiredMaxReplicas == nil {
+				maxReplicas = ng.MaxInstances
+			} else {
+				maxReplicas = *desiredMaxReplicas
+			}
+
+			if minReplicas < 0 {
+				return clusterconfig.Config{}, ErrorMinInstancesLowerThan(0)
+			}
+			if maxReplicas < 0 {
+				return clusterconfig.Config{}, ErrorMaxInstancesLowerThan(0)
+			}
+			if minReplicas > maxReplicas {
+				return clusterconfig.Config{}, ErrorMinInstancesGreaterThanMaxInstances(minReplicas, maxReplicas)
+			}
+
+			if ng.MinInstances == minReplicas && ng.MaxInstances == maxReplicas {
+				fmt.Printf("no changes to the %s nodegroup required in cluster %s in region %s\n", ng.Name, clusterName, region)
```

Comment:
nit:

The %s nodegroup in the %s cluster in %s already has min instances set to %d and max instances set to %d